_posts/2019-02-06-Age-Targeting-Part2.md

```diff
@@ -0,0 +1,253 @@
+|55+      |2.08  |solitarie card games                           |
+|55+      |2.04  |solitarie card games                          |
+
+As you can see the best features of each class are the apps used.
+Due to the bag of words preprocess on this feature, every apps is a distinct input, so the weights in the table are directly related to the discrimination power derived by the usage of that app.
+Moreover, the list makes extreme sense based on naive expectation.
+Solving your math homework is mostly a task you want to support with an app when you are young.
+In certain periods of your life you will probably do big expense (e.g. an house or a boat) and you are more likely to check your financial balance with more attention.
+Games and sport will still be with you during your mature age and at some point you will have enough patience to appreciate the old classic card games. 
+This gives additional confidence in the behavior of the model.
+
+Analyzing the output of the model is very helpful, especially in the cases it is wrong.
+In the plot you can see the distribution of the difference between the real age and the estimated age_group.
+Keep in mind that every class spans about ten years.
+Therefore, the range from -10 to 10 corresponds to adjacent misclassified classes.
+For example, if the age of the device is 36 and the model predicts the class 25_34 we have a distance of 36 -34 = 2 ∈ [-10, 10].
+Big mistakes, spanning more than one class occur rarely in comparison to the adjacent-like mistakes. 
+<p align="center"> <img src="../images/Age-Targeting/diff_age_agegroup.png"> </p>
+When we analyze those devices we see that their app usage would be commonly associated to the other classes.
+Well, in the end it is not important how old you are, but how old you feel.
+
+
+**Prediction over the entire dataset**<br/>
+The model gives as output the five probabilities for a device to belong to each of the age group categories.
+We use those probabilities in the ground truth sample to check the quality of the training and in the rest of the data to estimate the age of the users.
+In the plot below you can see the distribution for these two datasets for one of the five probabilities.
+We can see that the shape of the two distribution are different.
+<p align="center"> <img src="../images/Age-Targeting/output_predict_dist.png"> </p>
+These differences are not necessarily an indication for overfitting (that can be excluded because in the test set the distributions are very similar).
+In fact, the ground truth is a special part of the entire dataset because it contains the devices which visit the apps that provide the age information.
+With other words, the ground truth has a bias with respect to to the final dataset where the model will be applied.
+To fix the bias we reweight the ground truth distribution using the shape from the other dataset.
+The reweight is performed separately for each country because the output distribution is country-dependent.
```

Comment:
Since the output distribution is country-dependent, the reweighting is performant separately for each country.